render/render.js

```diff
@@ -161,86 +161,215 @@ module.exports = function($window) {
 	}
 
 	//update
-	function updateNodes(parent, old, vnodes, recycling, hooks, nextSibling, ns) {
-		if (old === vnodes || old == null && vnodes == null) return
+	/**
```

Comment:
I *seriously* appreciate this nice comment. I'm familiar with the argument order from having toyed with it for a while, but this is a nice little explanatory landmark that will make our lives *much* easier.